src/mame/namco/namcos12.cpp

```diff
@@ -1128,10 +1133,12 @@ class namcos12_state : public driver_device
 		, m_maincpu(*this, "maincpu")
 		, m_ram(*this, "maincpu:ram")
 		, m_sub(*this, "sub")
+		, m_cdxa_pcb(*this, "cdxa_pcb")
 		, m_adc(*this, "sub:adc")
 		, m_rtc(*this, "rtc")
 		, m_settings(*this, "namco_settings")
 		, m_sharedram(*this, "sharedram")
+		, m_cdxa_ata(*this, "cdxa_pcb:ata")
```

Comment:
Can you use a derived class for the games with the CDXA board, so it doesn’t need to be optional?  Also, as previously mentioned, messing with devices that aren’t immediate children (in this case that ATA interface) is usually a sign of a leaky abstraction.

Reply:
A derived class just for this feels overkill. I wasn't sure what to do about it when I wrote it but looking at it now I think exposing a cdrom_r/cdrom_w from namcos12_cdxa_device would be enough.